### ISSUE TITLE
[DynamicType] Tracking filed compiler bugs

### ISSUE THREAD

**Issue by zasdfgbnm** [NVIDIA/Fuser#914]
dynamic_type lib does not compile on C++20 even with the latest g++. I filed two compiler bugs:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111314
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111316
awaiting confirmation.

dynamic_type does compile on C++20 with clang++, but the test is failing for comparison operators of containers. Not sure if this is a compiler bug or a standard library bug, or neither. I narrowed it down to a minimal repro that is reproducible both on g++ and clang++. Trying at:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111504
https://github.com/llvm/llvm-project/issues/67056

**Comment by kevinstephano** [comment]
@zasdfgbnm with our switch over to C++20, is this issue still revelant?

**Comment by zasdfgbnm** [comment]
Yes, this is still relevant. But it is not something that we should do. I will move it to discussions.

**Comment by zasdfgbnm** [comment]
https://github.com/NVIDIA/Fuser/discussions/4963
